### ISSUE TITLE
OSS does not work with nvidia apex 

### ISSUE THREAD

**Issue by benywon** [facebookresearch/fairscale#270]
## ❓ Questions and Help
Before fairscale, I usually use the Nvidia apex for mixed-precision training. However, when I wrap the optimizer with OSS, it seems that the loss never decrease. Below is some snapshot of my code:
------before----
optimizer = apex.optimizers.FusedAdam(
    model.parameters(),
    weight_decay=0.01,
    eps=1.0e-7,
    lr=1.0e-7,
)
model, optimizer = amp.initialize(model,
                                  optimizer,
                                  opt_level="O2",
                                  loss_scale='dynamic',
                                  keep_batchnorm_fp32=False,
                                  verbosity=0
                                  )
model = apex.parallel.DistributedDataParallel(model)

for somesteps in training:
    loss = model([x.cuda(non_blocking=True) for x in one])
    with amp.scale_loss(loss, optimizer) as scaled_loss:
        scaled_loss.backward()
    optimizer.step()
    optimizer.zero_grad()

-------------after adding OSS to wrap the optimizer with only one line code------------------
optimizer = OSS(params=model.parameters(), optim=apex.optimizers.FusedAdam,
                weight_decay=0.01,
                eps=1.0e-7,
                lr=1.0e-9)
model, optimizer = amp.initialize(model,
                                  optimizer,
                                  opt_level="O2",
                                  loss_scale='dynamic',
                                  keep_batchnorm_fp32=False,
                                  verbosity=0
                                  )
model = apex.parallel.DistributedDataParallel(model)

for somesteps in training:
    loss = model([x.cuda(non_blocking=True) for x in one])
    with amp.scale_loss(loss, optimizer) as scaled_loss:
        scaled_loss.backward()
    optimizer.step()
    optimizer.zero_grad()
---------------------
and after adding the code, I got the following warning:
`WARNING:root:Assuming single node job, bucketing is disabled
WARNING:root:Optimizer state has not been consolidated. Returning the local state
WARNING:root:Please call `consolidate_state_dict()` beforehand if you meant to save the global state`

and the loss never decreases. So how should I wrap this OSS with the apex optimizer and mixed-precision package?

Thanks!!!

**Comment by blefaudeux** [comment]
Hi Bingning, thanks for writing that up ! I should make it clearer in the docs, but OSS is not compatible with Apex AMP, it would require upstream changes (the scaler is not aware of the sharding basically) and Apex AMP is in the process of being deprecated. You can however use Pytorch AMP to similar effect if you replace the GradScaler by the provided ShardedGradScaler, see https://fairscale.readthedocs.io/en/latest/

**Comment by benywon** [comment]
Ok, thanks for your reply!
